### DISCUSSION TITLE
Official inline documentation

### DISCUSSION THREAD

**Discussion by jonathanwiesel** [forcedotcom/salesforcedx-vscode#5790] [Ideas]
**Is your feature request related to a problem? Please describe.**
As a developer I constantly find myself looking at the official docs for specifics or detail for certain methods or components. Taking into account that all coding takes place inside VSCode, you need to go to the browser, search for the doc type (Apex, Aura, Visualforce, LWC), search for the class / component, then finally for the method or specific for that; this represents losing time and content switching.

**Describe the solution you'd like**
There are solutions for direct access to documentation right from VSCode (even with the selected keyword as context) using an integration with [Dash](https://kapeli.com/dash) / [Zeal](https://zealdocs.org/) such as [this one](https://github.com/deerawan/vscode-dash). This solutions however requires offline docset to be generated and loaded into the app, if the Salesforce team could officially support this format it could be loaded into these tools and benefit from a great user experience regarding documentation exploration right from the editor.

**Describe alternatives you've considered**
There are some developers that have created scripts to generate these docsets scraping the official docs such as the following ones

https://github.com/jairzh/sfdc-dash-docsets

https://github.com/ViViDboarder/docset-sfdc

These however are incomplete or old, and don't support all the Salesforce languages / frameworks (Apex, LWC, Visualforce, Aura).

**Comment by jonathanwiesel** [comment]
Since the main objective of this would be to directly access Salesforce official classes and components documentation directly inside VS Code maybe there are other approaches for the same goal. 

Something native that occurs to me is to directly make it part of the Salesforce language server extensions, although I'm no expert.

**Comment by jonathanwiesel** [comment]
Actually it could be part of the Intellisense capabilities implemented in the Apex extension, specifically by the [**quick info** functionality](https://code.visualstudio.com/docs/editor/intellisense#_intellisense-features). The UX would be awesome and native

**Comment by Oblongmana** [comment]
@jonathanwiesel 
Having it as part of the Intellisense stuff would be the ideal, though those pop-ups don't fill every need - there's a lot of useful doc that either isn't neatly tied to things like a class or method, or is painfully long-winded.

The 10 second wait for the SF Angular app that loads and renders doc is maddening enough though, that I made an extension that uses the underlying ToC etc from the SF docs to show you a quick pick in VSCode for searching headers in a variety of doc types, and either takes you direct to that piece of documentation (e.g. a specific method), or with an experimental flag on, it can load the chunk of doc in a VSCode view. Doesn't work offline, but caches the ToC so searches are quick. https://marketplace.visualstudio.com/items?itemName=Oblongmana.vscode-salesforce-doc-lookup

**Comment by jonathanwiesel** [comment]
@Oblongmana sounds promising, thanks a lot for the hard work, will definitely give it a try
